airflow/providers/amazon/aws/hooks/quicksight.py

```diff
@@ -154,18 +154,20 @@ def wait_for_state(
         :return: response of describe_ingestion call after Ingestion is is done
         """
         sec = 0
-        status = self.get_status(aws_account_id, data_set_id, ingestion_id)
-        while status in self.NON_TERMINAL_STATES and status != target_state:
+        while True:
+            status = self.get_status(aws_account_id, data_set_id, ingestion_id)
             self.log.info("Current status is %s", status)
             if status in self.FAILED_STATES:
                 info = self.get_error_info(aws_account_id, data_set_id, ingestion_id)
                 raise AirflowException(f"The Amazon QuickSight Ingestion failed. Error info: {info}")
             if status == "CANCELLED":
                 raise AirflowException("The Amazon QuickSight SPICE ingestion cancelled!")
-            # wait and try again
+            if status not in self.NON_TERMINAL_STATES or status == target_state:
+                break
             time.sleep(check_interval)
-            sec += check_interval
-            status = self.get_status(aws_account_id, data_set_id, ingestion_id)
+        # wait and try again
+        sec += check_interval
+        status = self.get_status(aws_account_id, data_set_id, ingestion_id)
```

Comment:
Not only is the indenting wrong, but I don't think we need any of these lines: `sec` is never used and `status` is set in the while loop
```suggestion
```

Reply:
+1